### PR TITLE
Make transport parameters varint

### LINKED ISSUE #3294: Make transport parameter ID and length varint
Now that we have turned almost every integer in the spec to varint, and we have removed any mention of the TLS presentation language, it makes sense to make transport parameters varint as well. The current uint16-based encoding sticks out like a sore thumb. That said, this change is more than cosmetic - transport parameters are our main joint for extending QUICv1, and switching to 62-bit will ensure we do not run out of space, ever. If we do not do this now, we will not be able to change it later in QUICv1 as these are used during the handshake.

This issue was originally described in #3020 but the feedback in Cupertino was that #3020 was conflating three different topics:
1. the language used to describe transport parameters. That has been changed away from the TLS presentation language via merged PR #3108.
2. the policy for the transport parameter IANA registry. This has been changed by merged PR #3170.
3. the encoding of transport parameter ID and length over the wire. This is discussed in this issue and the associated open PR #3169.

From the discussions I've seen, it looks like there was some opposition and some support. I'm opening this issue now so we can have that conversation here independent of topics (1) and (2) above.

### PR DESCRIPTION
**Pull request by DavidSchinazi**
Now that transport parameters are represented using QUIC encoding (as opposed to TLS presentation language), the remaining 16-bit fields stick out like sore thumbs. This PR makes transport parameters consistent with the rest of the document.

Fixed #3294.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -4571,10 +4565,14 @@ preferred_address (0x000d):
 +                                                               +
 |                                                               |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+| CID Length (8)|
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: I think that this change is unnecessary, but you missed a varint.

Comment (DavidSchinazi) [reply draft-ietf-quic-transport.md]: Connection ID lengths use a 1-byte encoding throughout the document (long headers, NEW_CONNECTION_ID). I'd rather keep this consistent.

Comment (martinthomson) [reply draft-ietf-quic-transport.md]: Ahh, right.  In consistency FTW.

**Comment by martinthomson** [comment]
@DavidSchinazi, as we have decided not to do this, can we close this?  (It doesn't have an issue, so if we wanted to pursue it, we'd need one.)

**Comment by DavidSchinazi** [comment]
@martinthomson I don't think we've quite decided not to do this yet. The sense I got in the room in Cupertino was that some folks were opposed to changing this, but the main opposition was the fact that I had conflated three issues into one (TLS presentation language, IANA registry, varint). The first one #3108 was merged, the second one #3170 is almost merged. I was thinking of filing a new issue and rebasing this PR when #3170 gets merged to have a conversation specific to this change.

**Comment by dtikhonov** [comment]
My recollection matches that of @martinthomson.  Unfortunately, I don't see it in the IETF 106 minutes and I don't have it in my own notes (I wasn't documenting every issue we covered...)

**Comment by kazuho** [comment]
FWIW my recollection also matches that of @martinthomson, and the relevant minutes are at https://github.com/quicwg/wg-materials/blob/master/interim-19-10/minutes.md#issue-3020-transport-parameter-registry-changes. It was discussed in Cupertino.

**Comment by marten-seemann** [comment]
I agree with @DavidSchinazi. Now that we're not using TLS presentation language any more, there's no need to continue using uint16s.

**Comment by kazuho** [comment]
> I agree with @DavidSchinazi. Now that we're not using TLS presentation language any more, there's no need to continue using uint16s.

I disagree with that way of thinking, as it is my understanding that we agreed on moving away from TLS presentation language _based on the premise_ that we would be using 16-bit lengths.

**Comment by ianswett** [comment]
The notes are really sparse on the varint conversation, varint is mentioned 3 times.

Because this issue was conflated with other issues, it wasn't discussed much by itself.  I thought we decided to split it up, do the first two, and then discuss the third, which I remember some in the room commenting wasn't going to happen, but I didn't think that constituted consensus of any sort.

All that said, unless we have strong support for changing this, I think the status quo wins.

**Comment by DavidSchinazi** [comment]
I've now created issue #3294 so we can move the conversation there.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -4634,10 +4628,14 @@ preferred_address (0x000d):
...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                      Connection ID (*)                      ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md]: What's the rationale for reordering the contents of this TP payload in this PR?

Comment (rpaulo) [reply draft-ietf-quic-transport.md]: This came from https://github.com/quicwg/base-drafts/issues/2944 since they were both cosmetic.

Comment (DavidSchinazi) [reply draft-ietf-quic-transport.md]: Based on feedback from the room, I've reverted that change.

**Pull request merged**